### ISSUE TITLE
Implement scalability refactor for miner

### ISSUE THREAD

**Issue by juanmanso** [masa-finance/masa-bittensor#94]
## Description

After masa-finance/bt-refactor-spike#1, we are now able to put this refactor into action on the `masa-bittensor` repo.

TL;DR for planning:
- use `self.axon.attach()` for miner instead of what we have now
- define a way to generate synapses for each task
- follow the folder structure suggested 

Use as reference [the draft PR](https://github.com/masa-finance/masa-bittensor/pull/104) which spikes the implementation

## Acceptance criteria

- Regression testing proof showing that functionalities persist
- Goals of masa-finance/masa-bittensor#71 are met

**Comment by juanmanso** [comment]
Blocked by #93

**Comment by juanmanso** [comment]
Unblocked by [bt-refactor-spike](https://github.com/masa-finance/bt-refactor-spike/issues/1)

**Comment by juanmanso** [comment]
I think this is ready for development

**Comment by juanmanso** [comment]
> I think this is ready for development
> 
> EDIT: I think that triage here means that it might not be well defined. What I meant with the previous text is that I think this is defined enough so we can remove the triage label and as a team select it for ready either above or below the waterline

+1

**Comment by juanmanso** [comment]
> > I think this is ready for development
> > EDIT: I think that triage here means that it might not be well defined. What I meant with the previous text is that I think this is defined enough so we can remove the triage label and as a team select it for ready either above or below the waterline
> 
> +1

+1

**Comment by Luka-Loncar** [comment]
Closing this as duplicate of spike from bt-refactor-spike #7
